### PR TITLE
Fixes to mo-dev script

### PR DESCRIPTION
**Pull request by pellaea**
I've just made a bunch of changes to get the VM to install on my laptop.  It was a real mess this time.

(Not sure why it keeps breaking, why can't it just stay fixed for a change?? :)

1. I have to "apt-get install libgmp3-dev" package.

2. I have to "source ~/.rvm/scripts/rvm" *within* mo-dev.

3. It wasn't checking correctly if the version of ruby it wants to run in /vagrant/mushroom-observer is correct.

4. I have to "gem install bundle" (not bundler, and for some reason I have to do it either as me or within /vagrant/mushroom-observer, don't ask).

I think that's it.

Anyone want to take a look at the changes before I merge it?  I don't want to break it for everyone else.  I tried to make all the changes only do something if something was explicitly missing.

-Jason

### CODE REVIEW

**Comment by JoeCohen** [comment]
I'll have a look shortly.

On Thu, Jan 14, 2016 at 4:09 PM Jason Hollinger <notifications@github.com>
wrote:

> I've just made a bunch of changes to get the VM to install on my laptop.
> It was a real mess this time.
>
> (Not sure why it keeps breaking, why can't it just stay fixed for a
> change?? :)
>
>    1.
>
>    I have to "apt-get install libgmp3-dev" package.
>    2.
>
>    I have to "source ~/.rvm/scripts/rvm" *within* mo-dev.
>    3.
>
>    It wasn't checking correctly if the version of ruby it wants to run in
>    /vagrant/mushroom-observer is correct.
>    4.
>
>    I have to "gem install bundle" (not bundler, and for some reason I
>    have to do it either as me or within /vagrant/mushroom-observer, don't ask).
>
> I think that's it.
>
> Anyone want to take a look at the changes before I merge it? I don't want
> to break it for everyone else. I tried to make all the changes only do
> something if something was explicitly missing.
>
> -Jason
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/MushroomObserver/developer-startup/pull/18
> Commit Summary
>
>    - Bunch of things were missing or wrong on my (fresh) VM. The changes
>    to mo-dev should detect these problems and fix them.
>
> File Changes
>
>    - *M* mo-dev
>    <https://github.com/MushroomObserver/developer-startup/pull/18/files#diff-0>
>    (28)
>
> Patch Links:
>
>    - https://github.com/MushroomObserver/developer-startup/pull/18.patch
>    - https://github.com/MushroomObserver/developer-startup/pull/18.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MushroomObserver/developer-startup/pull/18>.
>
-- 
--*Joe*

**Comment by JoeCohen** [comment]
I can get it to run, but it didn't work as smoothly for me as the current master:
- while running mo-dev /vagrant (the first time) I see:
  - `ERROR 1045 (28000): Access denied for user 'mo'@'localhost' (using password: YES)` around line 131 in the attached file.  I don't think (but I'm not certain) I've seen this before.
  - `Gem::RemoteFetcher::FetchError: Errno::ETIMEDOUT: Connection timed out - connect(2) for "rubygems.org" port 443 (https://rubygems.org/gems/mime-types-2.99.gem)`  ~ line 164 Might be random.
 - At end of first run ([complete output here](https://github.com/MushroomObserver/developer-startup/files/91262/develop-startup.txt)):
```
An error occurred while installing mime-types (2.99), and Bundler cannot continue.
Make sure that `gem install mime-types -v '2.99'` succeeds before bundling.
Could not find activesupport-4.2.5 in any of the sources
Run `bundle install` to install missing gems.
Could not find activesupport-4.2.5 in any of the sources
Run `bundle install` to install missing gems.
Could not find activesupport-4.2.5 in any of the sources
Run `bundle install` to install missing gems.
RVM installed.  Run: source /home/vagrant/.rvm/scripts/rvm
```
  - I fixed by running `source /home/vagrant/.rvm/scripts/rvm`, then the [fix bundle-related error procedure in README.md](https://github.com/MushroomObserver/developer-startup#fix-bundle-related-error)

- Other:
  - You may need to patch Ruby on your the VM on your laptop.  See [here](https://groups.google.com/forum/#!searchin/mo-developers/patch/mo-developers/1HdwcwozhEs/9Eo3Ur5HCwAJ).
 - If you're going to tweak the VM, consider omitting the documentation for bundle and bundler

[develop-startup.txt]

  `

**Comment by pellaea** [comment]
Damn, I was being so careful.

re: remotefetcher error -- random, ignore

re: db access denied -- I got this once, too, but it went away before I
could fix it, not sure why it ever happened, maybe associated with some
build or env error earlier on?

re: bundle-related error -- Interesting, okay, so you get the same error.
I've gotten this repeatedly in the past, too.  I keep "fixing" it but it
keeps becoming unfixed.  This time around I fixed it for mine by including
"gem install bundle" near the end.  (And it only tells it to install it if
gem thinks it isn't installed.  That's why I was hoping it wouldn't break
it for anyone else.)

Question: Are you absolutely certain the old one works better for you?  Or
do you just remember it working better earlier, maybe weeks ago?  This used
to all work perfectly hands-off for me (and does again now).

What worked for me was to set "breakpoints" in mo-dev.  I ran each bit one
piece at a time, verifying that everything was what it was supposed to be,
e.g., ruby version, etc.  When it wasn't I added a check for the missing
dependency or whatever and inserted a fix if that dependency wasn't met.
Then I started from scratch: delete the VM and all associated files in
virtualbox console, then run "startup" script again.  Over and over(!)
Until it ran smoothly start to finish.

It suggests a possibility in a fictitious world in which you had infinite
time: First start from scratch and reinstall with the old script and ensure
that it really does run smoothly for you start to finish.  If so, start
adding in the modifications I added one at a time, each time starting over
from scratch, and see which changes I made are causing these mysterious db
access and bundler errors.

Maybe that would finally give us the insight we need to understand why your
(identical!) VM works differently than mine.  Arg!  It's got to be because
we're relying on some unstable installation procedure, but I'll be damned
if I can see what it is.

Or we can just drop the issue, include two (or more!) mo-dev versions with
developer-startup and tell users to try them all, and hopefully one will
work for them! :)

-Jason

**Comment by JoeCohen** [comment]
I'll try installing the current master tomorrow AM, and let you know if I
was imagining things.

On Thu, Jan 14, 2016 at 8:05 PM Jason Hollinger <notifications@github.com>
wrote:

> Damn, I was being so careful.
>
> re: remotefetcher error -- random, ignore
>
> re: db access denied -- I got this once, too, but it went away before I
> could fix it, not sure why it ever happened, maybe associated with some
> build or env error earlier on?
>
> re: bundle-related error -- Interesting, okay, so you get the same error.
> I've gotten this repeatedly in the past, too. I keep "fixing" it but it
> keeps becoming unfixed. This time around I fixed it for mine by including
> "gem install bundle" near the end. (And it only tells it to install it if
> gem thinks it isn't installed. That's why I was hoping it wouldn't break
> it for anyone else.)
>
> Question: Are you absolutely certain the old one works better for you? Or
> do you just remember it working better earlier, maybe weeks ago? This used
> to all work perfectly hands-off for me (and does again now).
>
> What worked for me was to set "breakpoints" in mo-dev. I ran each bit one
> piece at a time, verifying that everything was what it was supposed to be,
> e.g., ruby version, etc. When it wasn't I added a check for the missing
> dependency or whatever and inserted a fix if that dependency wasn't met.
> Then I started from scratch: delete the VM and all associated files in
> virtualbox console, then run "startup" script again. Over and over(!)
> Until it ran smoothly start to finish.
>
> It suggests a possibility in a fictitious world in which you had infinite
> time: First start from scratch and reinstall with the old script and ensure
> that it really does run smoothly for you start to finish. If so, start
> adding in the modifications I added one at a time, each time starting over
> from scratch, and see which changes I made are causing these mysterious db
> access and bundler errors.
>
> Maybe that would finally give us the insight we need to understand why your
> (identical!) VM works differently than mine. Arg! It's got to be because
> we're relying on some unstable installation procedure, but I'll be damned
> if I can see what it is.
>
> Or we can just drop the issue, include two (or more!) mo-dev versions with
> developer-startup and tell users to try them all, and hopefully one will
> work for them! :)
>
> -Jason
>
>
> On Thu, Jan 14, 2016 at 6:08 PM, Joseph D. Cohen <notifications@github.com
> >
> wrote:
>
> > I can get it to run, but it didn't work as smoothly for me as the current
> > master:
> >
> > - while running mo-dev /vagrant (the first time) I see:
> > - ERROR 1045 (28000): Access denied for user 'mo'@'localhost'
> > (using password: YES) around line 131 in the attached file. I don't
> > think (but I'm not certain) I've seen this before.
> > - Gem::RemoteFetcher::FetchError: Errno::ETIMEDOUT: Connection
> > timed out - connect(2) for "rubygems.org" port 443 (
> > https://rubygems.org/gems/mime-types-2.99.gem) ~ line 164 Might be
> > random.
> > - At end of first run (complete output here
> > <
> https://github.com/MushroomObserver/developer-startup/files/91262/develop-startup.txt
> >
> > ):
> >
> > An error occurred while installing mime-types (2.99), and Bundler cannot
> continue.
> > Make sure that `gem install mime-types -v '2.99'` succeeds before
> bundling.
> > Could not find activesupport-4.2.5 in any of the sources
> > Run `bundle install` to install missing gems.
> > Could not find activesupport-4.2.5 in any of the sources
> > Run `bundle install` to install missing gems.
> > Could not find activesupport-4.2.5 in any of the sources
> > Run `bundle install` to install missing gems.
> > RVM installed. Run: source /home/vagrant/.rvm/scripts/rvm
> >
> >
> > - I fixed by running source /home/vagrant/.rvm/scripts/rvm, then the fix
> > bundle-related error procedure in README.md
> > <
> https://github.com/MushroomObserver/developer-startup#fix-bundle-related-error
> >
> >
> >
> > - Other:
> > - You may need to patch Ruby on your the VM on your laptop. See here
> > <
> https://groups.google.com/forum/#!searchin/mo-developers/patch/mo-developers/1HdwcwozhEs/9Eo3Ur5HCwAJ
> >
> > .
> > - If you're going to tweak the VM, consider omitting the
> > documentation for bundle and bundler
> >
> > [develop-startup.txt]
> >
> > `
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/MushroomObserver/developer-startup/pull/18#issuecomment-171847105
> >
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MushroomObserver/developer-startup/pull/18#issuecomment-171866979>
> .
>
-- 
--*Joe*

**Comment by JoeCohen** [comment]
TL;DR:  I think you should merge your changes.
maybe: `sudo gem install bundle --no-rdoc --no-ri` (in 2 places) instead of: `sudo gem install bundle`
Also, with `rake lang:update`, I don't see Japanese. If this is a trivial fix, maybe do it, else ignore.

Longer version:
I'm certain that sometime recently, I installed from the then-current master.  I remember being surprised.
But this morning, things are different:
- I again tried installing from the current master.  It nows failed so spectacularly that the ordinary fix doesn't work.
- Then I installed from your branch and it worked perfectly.

**Comment by pellaea** [comment]
Okay, all's well that ends well.  I can only guess that things have
continued to drift even since you last installed successfully.  I still
have no idea why things drift, but there it is, they do.  I'll just merge
this latest batch (after addind the --no-rdoc --no-ri flags) and give up
trying to permanently fix this for everyone.  Apparently it's just
impossible. :(
